lib/web_ui/test/golden_tests/engine/canvas_draw_image_golden_test.dart

```diff
@@ -400,6 +409,158 @@ void testMain() async {
+      ..setRotationY(0.8)
+      ..setEntry(3, 2, 0.0005); // perspective
+    canvas.transform(transform.storage);
+    canvas.drawImage(createTestImage(), Offset(0, 100), new Paint());
+    await _checkScreenshot(canvas, 'draw_3d_image',
+        region: region,
+        maxDiffRatePercent: 6.0,
+        setupPerspective: true);
+  });
+
+  /// Regression test for https://github.com/flutter/flutter/issues/61245
+  test('Should render image with perspective inside clip area', () async {
+    final Rect region = const Rect.fromLTRB(0, 0, 200, 200);
+    final RecordingCanvas canvas = RecordingCanvas(region);
+    canvas.drawRect(region, Paint()..color = Color(0xFFE0E0E0));
+    canvas.translate(10, 10);
+    canvas.drawImage(createTestImage(), Offset(0, 0), new Paint());
+    Matrix4 transform = Matrix4.identity()
+      ..setRotationY(0.8)
+      ..setEntry(3, 2, 0.0005); // perspective
+    canvas.transform(transform.storage);
+    canvas.clipRect(region, ClipOp.intersect);
+    canvas.drawRect(Rect.fromLTWH(0, 0, 100, 200), Paint()..color = Color(0x801080E0));
+    canvas.drawImage(createTestImage(), Offset(0, 100), new Paint());
+    canvas.drawRect(Rect.fromLTWH(50, 150, 50, 20), Paint()..color = Color(0x80000000));
+    await _checkScreenshot(canvas, 'draw_3d_image_clipped',
+        region: region,
+        maxDiffRatePercent: 5.0,
+        setupPerspective: true);
+  });
+
+  test('Should render rect with perspective transform', () async {
+    final Rect region = const Rect.fromLTRB(0, 0, 400, 400);
+    final RecordingCanvas canvas = RecordingCanvas(region);
+    canvas.drawRect(region, Paint()..color = Color(0xFFE0E0E0));
+    canvas.translate(20, 20);
+    canvas.drawRect(Rect.fromLTWH(0, 0, 100, 40),
+        Paint()..color = Color(0xFF000000));
+    Matrix4 transform = Matrix4.identity()
+      ..setRotationY(0.8)
+      ..setEntry(3, 2, 0.001); // perspective
+    canvas.transform(transform.storage);
+    canvas.clipRect(region, ClipOp.intersect);
+    canvas.drawRect(Rect.fromLTWH(0, 60, 120, 40), Paint()..color = Color(0x801080E0));
+    canvas.drawRect(Rect.fromLTWH(300, 250, 120, 40), Paint()..color = Color(0x80E010E0));
+    canvas.drawRRect(RRect.fromRectAndRadius(Rect.fromLTWH(0, 120, 160, 40), Radius.circular(5)),
+        Paint()..color = Color(0x801080E0));
+    canvas.drawRRect(RRect.fromRectAndRadius(Rect.fromLTWH(300, 320, 90, 40), Radius.circular(20)),
+        Paint()..color = Color(0x80E010E0));
+    await _checkScreenshot(canvas, 'draw_3d_rect_clipped',
+        region: region,
+        maxDiffRatePercent: 1.0,
+        setupPerspective: true);
+  });
+
+  test('Should render color and ovals with perspective transform', () async {
+    final Rect region = const Rect.fromLTRB(0, 0, 400, 400);
+    final RecordingCanvas canvas = RecordingCanvas(region);
+    canvas.drawRect(region, Paint()..color = Color(0xFFFF0000));
+    canvas.drawColor(Color(0xFFE0E0E0), BlendMode.src);
```

Comment:
Let's also test `canvas.drawColor`/`drawPaint` with a transform to make sure we cover the whole picture.

Reply:
Added canvas_draw_color_tesst